Description: Docker is unsupported on kernels < 3.10, so this turns the warning into a fatal error (hard failure)
Forwarded: not-needed
Author: Tianon Gravi <admwiggin@gmail.com>
Reviewed-by: Paul Tagliamonte <paultag@debian.org>

Index: docker.io/daemon/daemon.go
===================================================================
--- docker.io.orig/daemon/daemon.go
+++ docker.io/daemon/daemon.go
@@ -1184,7 +1184,7 @@ func checkKernel() error {
 	} else {
 		if kernel.CompareKernelVersion(k, &kernel.KernelVersionInfo{Kernel: 3, Major: 10, Minor: 0}) < 0 {
 			if os.Getenv("DOCKER_NOWARN_KERNEL_VERSION") == "" {
-				logrus.Warnf("You are running linux kernel version %s, which might be unstable running docker. Please upgrade your kernel to 3.10.0.", k.String())
+				logrus.Fatalf("ERROR: You are running Linux kernel version %s, which is unsupported for running Docker. Please upgrade your kernel to 3.10+.", k.String())
 			}
 		}
 	}
